### PR TITLE
UF2 manipulation

### PR DESCRIPTION
**Pull request by JakubVanek**
This adds the UF2 manipulation code. Currently I only plan to implement the offline commands, as these cannot be emulated with other commands.

Fixes https://github.com/c4ev3/ev3duder/issues/15

### CODE REVIEW

**Comment by JakubVanek** [comment]
I have settled for the following interface:
```
ev3 uf2 pack <output uf2 file> <brick directory> <input files...>
ev3 uf2 unpack <input uf2 file> <output directory> [--with-paths]
```

No real directory mapping is done inside ev3duder to keep it generic. `<brick directory>` therefore has to be one of the directories supported by the `uf2d` daemon on the brick, i.e. `Projects`, `SD Card`, `USB Stick`, `Temporary Logs` or `Permanent Logs`.

File: src/funcs.h [hunk 1]
```diff
@@ -59,6 +59,11 @@ extern int listen_mode();
 
 extern int bridge_mode();
 
+//! Pack files into a UF2 archive
+static inline int uf2_pack(FILE *fp, const char *brickdir, const char **file_array, int files) { return -1; }
+
+//! Unpack files from a UF2 archive
+static inline int uf2_unpack(FILE *fp, const char *dstdir) { return -1; }
```
Comment (a3f) [inline src/funcs.h]: Prototype declared with two arguments here.

File: src/main.c [hunk 2]
```diff
@@ -497,6 +506,58 @@ int main(int argc, char *argv[])
...
+				}
+
+				fp = fopen(SANITIZE(argv[1]), "rb");
+				if (!fp)
+				{
+					errmsg = "error: cannot open UF2 file for reading";
+					ret = ERR_IO;
+					break;
+				}
+				ret = uf2_unpack(fp, SANITIZE(argv[2]), uf2_paths);
```
Comment (a3f) [inline src/main.c:684]: But used with three arguments here.

**Review by a3f — COMMENTED** [review]
Please fix and rebase on master and ensure all intermediate comments can be compiled without error.

**Comment by JakubVanek** [comment]
Thank you, I have completely missed that.. I have created https://github.com/c4ev3/ev3duder/pull/30 as a replacement.

**Pull request merged**